File: C/Game/Game.c
#include <time.h>
#include "../BoidWorld/BoidWorld.h"
#include "../BoidWorldRender/BoidWorldRender.h"

void Wait(long milliseconds)
{
    if (milliseconds <= 0) return;
    float inFloat = ((float)milliseconds) / 1000;
    WaitTime(inFloat);
}
int main(void)
{
    BoidWorld game = Game_Init();
    Draw_Init(&game);
    long timeStep = 20;
    while (!Draw_ShouldExit())
    {
        long time1 = clock();
        Game_Update(&game,timeStep);
        Draw_Update(&game);
        long time2 = clock();
        long diff = time2 - time1;
        long waitTime = timeStep - diff;
        Wait(waitTime);
    }
    Game_End(&game);
	return 0;
}

File: C#/Game/Game.cs
using System.Diagnostics;

public static class Game
{
    public static void Main()
    {
        var curentProcess = Process.GetCurrentProcess();
        var curentThread = Thread.CurrentThread;
        curentProcess.PriorityClass = ProcessPriorityClass.High;
        curentThread.Name = "MAIN";
        curentThread.Priority = ThreadPriority.Highest;
        var watch = new Stopwatch();

        var boidWorld = new BoidWorld();
        var windowThread = BoidWorldRender.Start(boidWorld);
        var timeStep = 20L;
        while (windowThread.IsAlive)
        {
            watch.Restart();
            boidWorld.Update(timeStep);
            watch.Stop();
            var diff = watch.ElapsedMilliseconds;
            var waitTime = timeStep - diff;
            Wait(waitTime);
        }
    }
    public static void Wait(long milliseconds)
    {
        if (milliseconds <= 0)
        {
            Console.WriteLine("SLOW");
            return;
        }
        Thread.Sleep((int)milliseconds);
    }
}